.github/workflows/tests.yml

```diff
@@ -225,3 +225,72 @@ jobs:
       - name: 'Run brakeman'
         run: |
           bundle exec rake brakeman
+
+  js_tests:
+    name: 'JavaScript tests'
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: 'postgres:11.5'
+        ports:
+          - '5432:5432'
+        options: '--health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5'
+      redis:
+        image: redis
+        ports:
+          - '6379:6379'
+        options: '--entrypoint redis-server'
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Set up gem cache
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gem-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gem-
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10.x'
+
+      - name: Set up yarn cache
+        id: yarn-cache
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Make sure Yarn is cached
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: 'Set up Ruby 2.6'
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 2.6
+
+      - name: 'Install PostgreSQL 11 client'
+        run: |
+          sudo apt-get -yqq install libpq-dev
+
+      - name: 'Install dependencies and set up databases'
+        env:
+          PGHOST: localhost
+          PGUSER: postgres
+          RAILS_ENV: test
+        run: |
+          yarn install
+          gem install bundler
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+          bin/rails db:setup
+          RAILS_ENV=test bin/rails webpacker:compile
```

Comment:
None of these commands and their Ruby version prerequisites should be necessary to run JS tests 🤔

Reply:
We need to run Webpacker, which depends on Rails, which depends on... everything

Reply:
Turns out we don't; we need webpacker for the JavaScript that we want to run in the browser. Jest however runs in Node, so we don’t need to compile or do anything to the code, it just runs natively.